File: vripper-server/src/main/java/tn/mnlr/vripper/SpringContext.java
package tn.mnlr.vripper;

import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.BeansException;
import org.springframework.context.ApplicationContext;
import org.springframework.context.ApplicationContextAware;
import org.springframework.context.ConfigurableApplicationContext;
import org.springframework.stereotype.Component;

import java.util.Map;

@Component
@Slf4j
public class SpringContext implements ApplicationContextAware {

  private static ConfigurableApplicationContext context;

  public static <T> T getBean(Class<T> beanClass) {
    return context.getBean(beanClass);
  }

  public static <T> Map<String, T> getBeansOfType(Class<T> beanClass) {
    return context.getBeansOfType(beanClass);
  }

  public static void close() {
    log.info("Application terminating...");
    if (context != null) {
      context.close();
    }
  }

  @Override
  public void setApplicationContext(ApplicationContext context) throws BeansException {

    // store ApplicationContext reference to access required beans later on
    SpringContext.context = ((ConfigurableApplicationContext) context);
  }
}

File: vripper-core/src/main/kotlin/me/mnlr/vripper/SpringContext.kt
package me.mnlr.vripper

import org.springframework.beans.BeansException
import org.springframework.context.ApplicationContext
import org.springframework.context.ApplicationContextAware
import org.springframework.context.ConfigurableApplicationContext
import org.springframework.scheduling.annotation.EnableScheduling
import org.springframework.stereotype.Component
import me.mnlr.vripper.delegate.LoggerDelegate

@Component
@EnableScheduling
class SpringContext : ApplicationContextAware {
    @Throws(BeansException::class)
    override fun setApplicationContext(context: ApplicationContext) {

        // store ApplicationContext reference to access required beans later on
        Companion.context = context as ConfigurableApplicationContext
    }

    companion object {
        private val log by LoggerDelegate()
        private lateinit var context: ConfigurableApplicationContext
        fun <T> getBean(beanClass: Class<T>): T {
            return context.getBean(beanClass)
        }

        fun <T> getBeansOfType(beanClass: Class<T>): Map<String, T> {
            return context.getBeansOfType(beanClass)
        }

        fun close() {
            log.info("Application terminating...")
            context.close()
        }
    }
}